### ISSUE TITLE
5.3 compatibility

### ISSUE THREAD

**Issue by balping** [thekordy/ticketit#243]
Hi @thekordy 

I started working on 5.3 compatibility. See zero-point-two-53 branch. The reason behind the branch name is that composer can't handle branch names starting with digits when you use local repos...

Anyway after a few hours of debugging yesterday I gave up and now I'm asking for your help. I just can't imagine what's happening. I'm not even sure if it's something wrong on our side, or laravel has just messed up something since 5.2.

The error occurs on installing ticketit. The exact command causing this is in InstallController, where it tries to run artisan vendor:publish. Then it gets into a some kind of infinite recursion (see my attached [file](https://github.com/thekordy/ticketit/files/429113/debebug.txt)). Increasing the nesting limit doesn't help - if you give it a high enough value, you'll get a segmentation fault instead of error messages. (I wouldn't thought that php can cause segfault actually, I've only run into this when I was coding in C, where you can easily mishandle memory).

So please look into this if you have time. Laravel 5.3 will be released on august 23 at laracon eu, if the rumors are true. I thought it would be nice if ticketit were 5.3 compatible at the time of the release, but of course nothing happens if not. Another option is not supporting 5.3 at all of course, maybe it's enough if v2 is 5.3 compatible. What do you think?

<hr>

PS. This is a convenient way of testing, I encourage you to follow my composer.json file. Hope this helps.

```
"require": {
        "php": ">=5.6.4",
        "laravel/framework": "5.3.*",
        "kordy/ticketit": "dev-zero-point-two-53"
    },
    "repositories": [
        {
            "type":"path",
            "url":"../ticketit"
        }
    ],
}
```

You can install laravel 5.3 beta by running this command by the way: `composer create-project laravel/laravel tickettest dev-develop`

**Comment by thekordy** [comment]
I will do my best

**Comment by thekordy** [comment]
weird .. any Artisan command will give `500` server error

``` php
// routes/web.php
Route::get('test', function (){
    Artisan::call('migrate');
   return 'hello';
});
```

this command or any other commands from what I've tried will give the `500` server error

**Comment by balping** [comment]
And when you inspect the error log, you get "Maximum function nesting level reached", right?

**Comment by balping** [comment]
I think it's a bug in laravel, I'll try to reduce the problem and make a report in a form of a test, if I succeed

**Comment by thekordy** [comment]
also in v5.3 we can get ride of the migrations publish altogether by using the new feature `loadMigrationsFrom` in the service provider `boot()`

``` php
$this->loadMigrationsFrom(__DIR__.'/Migrations');
```

then user may only needs to do one command

```
php artisan migrate
```

I did some quick modifications just as a concept prove, I will push it now

**Comment by balping** [comment]
Yes, correct, it's failing with a clean install. I'll report it now

**Comment by thekordy** [comment]
> And when you inspect the error log, you get "Maximum function nesting level reached", right?

Yes indeed

```
2016/08/22 23:47:25 [error] 5545#5545: *380 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Maximum function nesting level of '256' reached, aborting! in /var/www/ticketit02-53/laravel/vendor/laravel/framework/src/Illuminate/Container/Container.php on line 696

...
```

**Comment by balping** [comment]
https://github.com/laravel/framework/issues/14950

**Comment by balping** [comment]
It will be fixed in 5.3.1 as they say, so let's wait until then. I think it's easier to stay with the old artisan way because then we dont have to make version-based switches

**Comment by kebian** [comment]
It looks like they've resolved that problem now.

**Comment by balping** [comment]
@kebian Yes, it's true.

I tested the package and it seems to work now. But we still have to wait for laravelcollective/html's 5.3 release. Because it now installs only if `"minimum-stability": "dev"` is turned on, we cant expect it from a user to use this config in their composer.json file. I'm using Sibbell, so I'll get an email notification when they release it. Then, we can release as well.

**Comment by balping** [comment]
Laravel collective 5.3 will be released today or tomorrow

https://github.com/LaravelCollective/html/issues/220#issuecomment-241861761

**Comment by balping** [comment]
Laravelcollective is released.

I merged the changes to the 0.2 branch. Now it should work with 5.3

@thekordy I think it's time for a new release (bug fixes, French translation, L5.3 compatibility)

**Comment by thekordy** [comment]
Great :+1:
